src/GearScalingOverWrite.h

```diff
@@ -24,19 +24,84 @@ extern "C" float GetGearScaling(cube::Item * item, cube::Creature* creature, int
 	float base_res = ((base * 0.5f) / (float)0x20);
 	float mod_modifier = (mod3 / 0x10624DD3) / 7.0f;
 
-	float X = 2;
+	float X = 1.4;
 	float Y = base_res + effective_rarity + mod_modifier;
 
 	float result = std::powf(X, Y);
 
 	cube::Game* game = cube::GetGame();
 	if (game && creature->entity_data.hostility_type == cube::Creature::EntityBehaviour::Player)
 	{
-		result *= 1 + 0.05f * GetItemLevel(item);
+		result *= 0.5 * std::pow(2.7183, 0.2 * GetItemLevel(item)) / 1.21 * std::pow(0.99, 1 + 0.07 * GetItemLevel(item) * GetItemLevel(item));
+
 	}
 	return result;
 }
 
+
+extern "C" float GetOtherStatsRe(cube::Item * item, cube::Creature * creature)
+{
+	IntVector2 region;
+	if (creature->entity_data.hostility_type == 0)
+	{
+		region = creature->entity_data.current_region;
+	}
+	else
+	{
+		region = item->region;
+	}
+
+	int mod = item->modifier;
+
+	int mod1 = mod ^ (mod << 0x0D);
+	int mod2 = mod1 ^ (mod1 >> 0x11);
+	int mod3 = mod2 ^ (mod2 << 0x05);
+
+	int effective_rarity = item->GetEffectiveRarity(&region);
+	float mod_modifier = (mod3 / 0x10624DD3) / 7.0f;
+	float randomizer = 2 * PyroRand(mod3) / 32768;
+
+	float X = 1.2;
+	float Y = effective_rarity + mod_modifier;
+
+	float result = std::powf(X, Y);
+
+	cube::Game* game = cube::GetGame();
+	if (game && creature->entity_data.hostility_type == cube::Creature::EntityBehaviour::Player)
+	{
+		result *= 0.01f + 0.0016f * GetItemLevel(item);
+		if (result > 0.2f) {
+			result = std::log2f(result/0.2f)*0.2f + 0.2f;
+			result *= randomizer;
+		}
+	}
+	else {
+		result = 0;
```

Comment:
This means that gear from creatures will have 0 haste, crit, and regen. Be aware of that effect. Can you explain the reasoning behind this choice?